### PR TITLE
add matching kubernetes secrets to environment during test execution

### LINKED ISSUE #25: Make secrets available as environment variables during test execution
Currently if someone wants to use a secret (e.g. API token) in a test, it would have to be hard coded into the tests and therefore not really be a secret anymore. Providing a way to inject respective secrets into a test would help to solve this problem. The test code can then expect the secrets to be available as environment variables.

As secret provider the keptn secret service would be the intended provider but it currently does not have a GET endpoint to retrieve the created secrets, services need to retrieve the secrets with the k8s resource api. Until the secret service provides a GET endpoint, the locust service can use secrets created in its namespace (keptn) as source and access them with the help of the k8s api.

The secrets could be a single key/value pair and have a naming pattern like `<project>-<stage>-<service>-<secret_name>`. When a test executes, the locust service can retrieve all matching secrets with the names of the project/stage/service and set them as environment variables without the prefix `<project>-<stage>-<service>-<secret_name> => <secret_name>`

### PR DESCRIPTION
**Pull request by augustin-dt**
Adds the possibility to make kubernetes secrets available as environment variables in locust tests.

implements #25

### CODE REVIEW

File: pkg/environment/environment_provider.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	log.Printf("Prepare data of secret %s as environment", secretName)
+
+	secret, err := e.KubeAPI.Secrets(e.KeptnNamespaceProvider()).Get(context.TODO(), secretName, metav1.GetOptions{})
+	if err != nil {
+		log.Printf("Unable to get secret %s: %s", secretName, err)
+		return environment
+	}
+
+	for key, value := range secret.Data {
+		environment = append(environment, fmt.Sprintf("%s=%s", key, string(value)))
```
Comment (christian-kreuzberger-dtx) [inline pkg/environment/environment_provider.go:41]: I think here we need to be careful and ensure that values that contain `"` and `=` also work.

E.g., what if my secret looks like this:
```
ABC="my key=value"
```

File: pkg/environment/environment_provider.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+func NewEnvironmentProvider(kubeAPI v1.CoreV1Interface) *EnvironmentProvider {
+	return &EnvironmentProvider{
+		KubeAPI:                kubeAPI,
+		KeptnNamespaceProvider: envBasedStringSupplier("POD_NAMESPACE", defaultNamespace),
+	}
+}
+
+func (e EnvironmentProvider) PrepareEnvironment(project string, stage string, service string) []string {
```
Comment (christian-kreuzberger-dtx) [inline pkg/environment/environment_provider.go:29]: Please add a comment here, e.g.,
```
// PrepareEnvironment creates a list of environment variables by extracting them from a secret based on project, stage and service
```

**Review by christian-kreuzberger-dtx — CHANGES_REQUESTED** [review]
Looks good, had one remark though

File: pkg/environment/environment_provider.go (hunk 1 above)
Comment (augustin-dt) [reply pkg/environment/environment_provider.go:41]: I refactored the tests to make sure the generated environment is correct in the context of a command execution. Also added a test for more complex environment variables.

File: pkg/environment/environment_provider.go (hunk 2 above)
Comment (augustin-dt) [reply pkg/environment/environment_provider.go:29]: I refactored the tests to make sure the generated environment is correct in the context of a command execution. Also added a test for more complex environment variables.

File: pkg/environment/environment_provider_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+	secretData := map[string][]byte{key1: []byte(value1), key2: []byte(value2)}
+	k8sSecret := createK8sSecretObj("locust-project-stage-service", namespace, secretData)
+	kubernetes.CoreV1().Secrets(namespace).Create(context.TODO(), k8sSecret, metav1.CreateOptions{})
+
+	environment := environmentProvider.PrepareEnvironment("project", "stage", "service")
+	assert.NotNil(t, environment)
+	assertEnvironmentVariable(t, environment, key1, value1)
+	assertEnvironmentVariable(t, environment, key2, value2)
+}
```
Comment (christian-kreuzberger-dtx) [inline pkg/environment/environment_provider_test.go:83]: Lovely! Thank you so much for this test!

**Review by christian-kreuzberger-dtx — APPROVED** [review]
Awesome! LGTM

**Pull request merged**